libpoolprotocols/PoolManager.cpp

```diff
@@ -351,6 +351,16 @@ int PoolManager::setActiveConnection(unsigned int idx)
     return 0;
 }
 
+int PoolManager::setActiveConnection(std::string poolid)
+{
+    for (size_t i = 0; i < m_connections.size(); i++)
+    {
+        if (m_connections[i].PoolId() == poolid)
```

Comment:
Also, for sake of user simplicity, I would consider the implications of string casing.

Reply:
I could ignore case.

I'll add checks for duplicate connection ids
The miner_addconnection can stay as is as it gets a URI as parameter that can include a fragment.
I'll add the pool id option to the delete connection API.